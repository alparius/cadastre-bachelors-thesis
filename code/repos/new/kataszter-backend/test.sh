#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
