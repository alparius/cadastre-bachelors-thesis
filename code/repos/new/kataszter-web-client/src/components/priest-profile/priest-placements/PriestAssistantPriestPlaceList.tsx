import React from "react";
import { Card } from "semantic-ui-react";

import IParishMin from "../../../data/parish/ParishMin";
import IPlacement from "../../../data/priest/Placement";
import IPriestProfileData from "../../../data/priest/PriestProfileData";
import "../../../static/priestProfile.css";
import ItalicFM from "../../../util/styledMessages/ItalicFM";
import PlacementCard from "./PlacementCard";

interface IPriestAssistantPriestPlaceListProps {
    priest: IPriestProfileData;
    editMode: boolean;
    parishResponse: IParishMin[] | undefined;
    parishLoading: boolean;
    parishShowError: boolean;
    setPriest(newPriest: IPriestProfileData): void;
}

const PriestAssistantPriestPlaceList: React.FC<IPriestAssistantPriestPlaceListProps> = (props: IPriestAssistantPriestPlaceListProps) => {
    const { priest, editMode, setPriest, parishResponse, parishLoading, parishShowError } = props;

    const handleDeletePlacement = (index: number) => {
        priest.assistantPriestPlaces.splice(index, 1);
        setPriest({
            ...priest,
            assistantPriestPlaces: priest.assistantPriestPlaces
        });
    };

    const handleChangeOnPlacementField = (labelName: string, value: any, index: number) => {
        setPriest({
            ...priest,
            assistantPriestPlaces: priest.assistantPriestPlaces.map((app, i) => (i === index ? { ...app, [labelName]: value } : { ...app }))
        });
    };

    const handlePlacementChange = (placeID: number, place: string, index: number) => {
        setPriest({
            ...priest,
            assistantPriestPlaces: priest.assistantPriestPlaces.map((app, i) => (i === index ? { ...app, placeID, place } : { ...app }))
        });
    };

    if (!priest.assistantPriestPlaces || priest.assistantPriestPlaces.length === 0) {
        return <ItalicFM id="general.noData" />;
    } else {
        return (
            <Card.Group stackable doubling itemsPerRow={2}>
                {priest.assistantPriestPlaces.map((assistantPriestPlace: IPlacement, index: number) => (
                    <PlacementCard
                        key={index}
                        index={index}
                        editMode={editMode}
                        placement={assistantPriestPlace}
                        handleDeletePlacement={handleDeletePlacement}
                        handleChangeOnPlacementField={handleChangeOnPlacementField}
                        handlePlacementChange={handlePlacementChange}
                        parishResponse={parishResponse}
                        parishLoading={parishLoading}
                        parishShowError={parishShowError}
                    />
                ))}
            </Card.Group>
        );
    }
};

export default PriestAssistantPriestPlaceList;
