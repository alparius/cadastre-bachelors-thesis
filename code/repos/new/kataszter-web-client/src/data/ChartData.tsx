interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
