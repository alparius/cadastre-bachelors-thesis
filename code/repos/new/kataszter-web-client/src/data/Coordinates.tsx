interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
