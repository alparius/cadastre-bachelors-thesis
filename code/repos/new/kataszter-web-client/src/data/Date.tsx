interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
