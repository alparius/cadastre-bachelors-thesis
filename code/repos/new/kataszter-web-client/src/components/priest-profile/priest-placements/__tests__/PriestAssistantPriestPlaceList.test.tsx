/*===================
  PriestAssistantPriestPlaceList.unit.test.tsx
 ===================*/
import React from "react";

import IPriestProfileData from "../../../../data/priest/PriestProfileData";
import ISpouse from "../../../../data/priest/Spouse";
import { NewTestPriestProfileData } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../../util/test/test-wrapper";
import PriestAssistantPriestPlaceList from "../PriestAssistantPriestPlaceList";

const handlers = {
    setPriest: jest.fn(),
    parishResponse: undefined,
    parishLoading: false,
    parishShowError: false
};

const priest: IPriestProfileData = NewTestPriestProfileData;

describe("PriestAssistantPriestPlaceList element", () => {
    it("should render a placement list container when the placement list is not empty", () => {
        const container = createComponentWithIntlAndRouter(
            <PriestAssistantPriestPlaceList priest={priest} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a `no data` when the placement list is empty", () => {
        const emptySpouses: ISpouse[] = [];
        priest.spouses = emptySpouses;

        const container = createComponentWithIntlAndRouter(
            <PriestAssistantPriestPlaceList priest={priest} editMode={false} {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });
});
