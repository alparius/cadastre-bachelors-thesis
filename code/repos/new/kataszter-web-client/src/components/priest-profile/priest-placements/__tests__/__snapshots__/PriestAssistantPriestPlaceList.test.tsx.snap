// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestAssistantPriestPlaceList element should render a \`no data\` when the placement list is empty 1`] = `
<div
  className="ui doubling stackable two cards"
>
  <a
    aria-current={null}
    className="ui fluid card"
    href="/parishes/profile/0"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="content"
      >
        <div
          className="header"
          id="header"
        >
          <h2
            className="ui header"
          >
            place
          </h2>
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="ui list"
          role="list"
        >
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Start date
                  :
                </b>
                 
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  2000.06.06
                </span>
              </div>
            </form>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  End date
                  :
                </b>
                 
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  2000.06.06
                </span>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`PriestAssistantPriestPlaceList element should render a placement list container when the placement list is not empty 1`] = `
<div
  className="ui doubling stackable two cards"
>
  <a
    aria-current={null}
    className="ui fluid card"
    href="/parishes/profile/0"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="content"
      >
        <div
          className="header"
          id="header"
        >
          <h2
            className="ui header"
          >
            place
          </h2>
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="ui list"
          role="list"
        >
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Start date
                  :
                </b>
                 
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  2000.06.06
                </span>
              </div>
            </form>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  End date
                  :
                </b>
                 
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  2000.06.06
                </span>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
